### ISSUE TITLE
newlib 2.5.0 and upstreaming

### ISSUE THREAD

**Issue by afaerber** [riscv/riscv-newlib#9]
As discussed at FOSDEM with @arunthomas, for Linux distros like [openSUSE](https://www.opensuse.org/) to pick up RISC-V toolchains, your target support needs to be in the respective upstream projects (and released tarballs).

At present, binutils 2.28 and gcc 7.0.1 do contain RISC-V target support, but apparently not yet [newlib](https://sourceware.org/git/?p=newlib-cygwin.git;a=summary), without which no riscv64-elf gcc cross-toolchain can be packaged by us yet, despite all other pieces being present.

Of course getting such things merged may take time, but there not being a 2.5.0 based branch here yet makes me doubt this is in progress already... Is anything in particular holding this up? I thought only glibc and Linux are blocked by the non-final Privileged ISA Specification, which should not impact, e.g., the HiFive1 board that newlib would be used for?

More general, https://riscv.org/software-tools/risc-v-gnu-compiler-toolchain/ does not seem to textually track the upstreaming progress anywhere, in terms of links to patchsets or done vs. to-do lists? It exclusively refers to your GitHub based projects, no direct instructions for how to configure the official tarballs where upstreaming is completed, and the last toolchain related news items were in early 2015. As convenient as your submodule projects may be for early RISC-V users building toolchains themselves, please understand that your submodules are not suitable for distribution and that they hide important configuration details in your riscv-gnu-toolchain Makefile.in and configure.ac files that would be really useful to clearly state somewhere on the official RISC-V site.

**Comment by kito-cheng** [comment]
I am working for newlib, I plan this month to clean up and then send to upstream.
and here is my working tree for newlib 2.5[1]
[1] https://github.com/kito-cheng/riscv-newlib/tree/riscv-newlib-2.5.0

**Comment by palmer-dabbelt** [comment]
Thanks for taking care of upstreaming this, Kito.  Do you mind putting your code on riscv/riscv-newlib, so everyone can find it more easily?

**Comment by kito-cheng** [comment]
It's ok but I don't run all testing yet, I will push to this repo once I finish testing.

**Comment by kito-cheng** [comment]
status update:
newlib 2.5 is merged into this repo, and it's under clean up

https://github.com/riscv/riscv-newlib/tree/riscv-newlib-2.5.0

**Comment by aswaterman** [comment]
cool

**Comment by afaerber** [comment]
Builds like 2.4 (i.e., riscv64-elf), thank you very much.
Please update us when there's progress with upstreaming.

**Comment by kito-cheng** [comment]
newlib is upstreaming[1] now :)

[1] https://sourceware.org/ml/newlib/2017/msg00672.html

**Comment by kito-cheng** [comment]
newlib is upstreamed.
https://groups.google.com/a/groups.riscv.org/forum/#!msg/sw-dev/XLO_iHwEfCU/XaeGX6RyAAAJ
